executor/builder.go

```diff
@@ -4037,6 +4045,30 @@ func partitionPruning(ctx sessionctx.Context, tbl table.PartitionedTable, conds
 	return ret, nil
 }
 
+func getPartitionIdsAfterPruning(ctx sessionctx.Context, tbl table.PartitionedTable, conds []expression.Expression, partitionNames []model.CIStr,
+	columns []*expression.Column, columnNames types.NameSlice) (map[int64]struct{}, error) {
+	idxArr, err := plannercore.PartitionPruning(ctx, tbl, conds, partitionNames, columns, columnNames)
```

Comment:
Here is not really partition pruning, just to filter out the selected partition?

Reply:
> Here is not really partition pruning, just to filter out the selected partition?
I don't know how to just filter out the selected partition gracefully. Is there any problems here by calling `plannercore.PartitionPruning`? 
I implement the `getPartitionIdsAfterPruning` by simulating `partitionPruning` method and this method is used in several places in builder.go. Why other `partitionPruning`s are good?